File: Backend/Entity_Layer/Dtos/RegisterUser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entity_Layer.Dtos
{
    public class RegisterUser
    {
        public string UserName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
        public Sex Sex { get; set; }
        public RelationshipStatus RelationshipStatus { get; set; } = RelationshipStatus.NotMentioned;
        public string Email { get; set; }
        public string Password { get; set; }
        public string ConfirmPassword { get; set; }
    }
}

File: Frontend/src/app/core/models/registerUser.ts
export class RegisterUser {
    constructor(
        public userName: string,
        public firstName: string,
        public lastName: string,
        public dateOfBirth: Date,
        public sex: number,
        public relationshipStatus: number,
        public email: string,
        public password: string,
        public confirmPassword: string,
    ) {}
}